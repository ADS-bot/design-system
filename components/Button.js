import React from 'react'

import { space, width, fontSize, color } from 'styled-system'
import styled from 'styled-components'

export const Button = styled.button`
  $space;
  $width;
  $fontSize;
  $color;


  // Common

  border: none; // maybe change
  border-radius: ${props => `${props.theme.radius}px`};
  display: inline-block;
  height: 36px;
  line-height: 36px;
  padding: 0 2rem;
  text-transform: uppercase;
  vertical-align: middle;
  // Gets rid of tap active state
  -webkit-tap-highlight-color: transparent;

  font-size: 1em;
  outline: 0;

  // Specific
  font-family: inherit;
  font-weight: 700;
  text-decoration: none;
  color: ${props => props.theme.colors.white};
  background-color: ${props => props.theme.colors.blue5};

  text-align: center;
  letter-spacing: .5px;
  z-index: 1;
  transition: .2s ease-out;
  cursor: pointer;

  &:hover {
    background-color: ${props => props.theme.colors.blue6};
    @extend .z-depth-1-half;
  }
`
export default Button
