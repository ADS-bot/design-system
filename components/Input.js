import React from 'react'

import { space, width, fontSize, color } from 'styled-system'
import styled from 'styled-components'

export const Input = styled.input`
  $space;
  $width;
  $fontSize;
  $color;

  font-family: inherit;
  font-size: 1rem;
  display: 'block';
  vertical-align: 'middle';

  background-color: transparent;

  border: none;
  border-bottom: 1px solid ${props => props.theme.colors.gray2};
  border-radius: 0;
  margin: 0;
  margin-bottom: 15px;
  padding: 0;

  outline: none;
  width: 100%;
  height: 2rem;

  box-shadow: none;
  box-sizing: content-box;
  transition: all .3s;

  &:focus:not([readonly]) {
    border-bottom: 1px solid ${props => props.theme.colors.base};
    box-shadow: 0 1px 0 0 ${props => props.theme.colors.gray1};
  }
`
export default Input
