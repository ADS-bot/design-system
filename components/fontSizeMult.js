/* global require */
import {
  system
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return system({
    'fontSize': {
      property: propName,
      scale: 'fontSizes',
      transform: n => n * k,
      defaultScale: defaultFontSizes
    }
  })
}

export default fontSizeMult
