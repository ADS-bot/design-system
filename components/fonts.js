import { injectGlobal } from 'styled-components'

import CharterRegular from '../fonts/Charter-Regular.woff'
import CharterBold from '../fonts/Charter-Bold.woff'

import FiraSansLight from '../fonts/FiraSans-Light.woff'
import FiraSansRegular from '../fonts/FiraSans-Regular.woff
import FiraSansBold from '../fonts/FiraSans-Bold.woff'

import SourceCodeProBold from '../fonts/SourceCodePro-Bold.woff'
import SourceCodeProRegular from '../fonts/SourceCodePro-Regular.woff'


injectGlobal`
  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansLight}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Fira Sans";
    src: url('${FiraSansSemiBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProRegular}') format('woff');
    font-style: normal;
    font-weight: 400;
  }

  @font-face {
    font-family: "Source Code Pro";
    src: url('${SourceCodeProBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterRegular}') format('woff');
    font-style: normal;
    font-weight: 300;
  }

  @font-face {
    font-family: "Charter";
    src: url('${CharterBold}') format('woff');
    font-style: normal;
    font-weight: 700;
  }
`
