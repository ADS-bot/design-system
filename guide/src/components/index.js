import  ColorPalette from './ColorPalette'

const GuideComponents = Object.assign({}, { ColorPalette })
export default GuideComponents
