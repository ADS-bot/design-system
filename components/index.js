import {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

import { Select, Label } from '@rebass/forms/styled-components'

import { ThemeProvider as Provider } from 'styled-components'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Flex,
  Box,
  Link,
} from 'rebass/styled-components'

export { Select, Label } from '@rebass/forms/styled-components'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Input,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, {
  Provider,
  Box,
  Flex,
  Link,
  Select,
  Label,
  colors,
  theme
})

export default OONIComponents
