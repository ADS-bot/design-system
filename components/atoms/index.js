import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
