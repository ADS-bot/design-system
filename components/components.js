import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */
import Button from './Button'
export { default as Button } from './Button'
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Container = atoms.Container

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass/styled-components'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Lead,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Small,
  Star,
  Sticky,
  Subhead,
  TabItem,
  Tabs,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
