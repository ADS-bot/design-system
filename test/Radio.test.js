/* global describe, test, expect */
import React from 'react'
import { renderWithTheme } from './index'
import { RadioButton, RadioGroup } from '../components'

describe('RadioButton', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <RadioButton />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with prop checked', () => {
    const { container } = renderWithTheme(
      <RadioButton checked />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})


describe('RadioGroup', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <RadioGroup>
        <RadioButton />
      </RadioGroup>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
