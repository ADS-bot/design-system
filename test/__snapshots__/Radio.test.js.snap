// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  cursor: pointer;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  font-size: 16px;
  cursor: pointer;
  padding-bottom: 16px;
}

.c5 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
}

.c6 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  border: 2px solid #adb5bd;
  background-color: 'transparent';
  position: absolute;
  top: 0;
}

<div
  class="c0"
  font-size="1"
>
  <label
    class="c1"
    font-size="1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          type="radio"
        />
        <div
          class="c5"
          font-size="1"
        >
          <div
            class="c6"
            font-size="1"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <div />
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton renders with prop checked 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  cursor: pointer;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  font-size: 16px;
  cursor: pointer;
  padding-bottom: 16px;
}

.c5 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
}

.c6 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  border: 2px solid #0588cb;
  background-color: 'transparent';
  position: absolute;
  top: 0;
}

.c7 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  border: 2px solid #0588cb;
  background-color: 'transparent';
  position: absolute;
  top: 0;
  background-color: #0588cb;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

<div
  class="c0"
  font-size="1"
>
  <label
    class="c1"
    font-size="1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          class="c4"
          type="radio"
        />
        <div
          class="c5"
          font-size="1"
        >
          <div
            class="c6"
            font-size="1"
          />
          <div
            class="c7"
            font-size="1"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <div />
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioGroup renders 1`] = `
.c3 {
  box-sizing: border-box;
}

.c2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  cursor: pointer;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 0;
  height: 0;
  margin: 0;
  opacity: 0;
  padding: 0;
  position: absolute;
  width: 0;
}

.c0 {
  font-size: 16px;
  cursor: pointer;
  padding-bottom: 16px;
}

.c5 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
}

.c6 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  border: 2px solid #0588cb;
  background-color: 'transparent';
  position: absolute;
  top: 0;
}

.c7 {
  position: relative;
  margin: 0;
  margin-right: 10px;
  vertical-align: middle;
  cursor: pointer;
  display: inline-block;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  border: 2px solid #0588cb;
  background-color: 'transparent';
  position: absolute;
  top: 0;
  background-color: #0588cb;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

<div>
  <div
    class="c0"
    font-size="1"
  >
    <label
      class="c1"
      font-size="1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <input
            class="c4"
            type="radio"
          />
          <div
            class="c5"
            font-size="1"
          >
            <div
              class="c6"
              font-size="1"
            />
            <div
              class="c7"
              font-size="1"
            />
          </div>
        </div>
        <div
          class="c3"
        >
          <div />
        </div>
      </div>
    </label>
  </div>
</div>
`;
